@chia7712 here's a run with the latest changes https://github.com/apache/kafka/actions/runs/10850348751?pr=17181

The "Parse JUnit Tests" step produces the following stdout:

```
Gradle command timed out. These are partial results!
1026 tests cases run in 41s. 1026 PASSED ✅, 0 FAILED ❌, 0 FLAKY ⚠️ , 0 SKIPPED 🙈, and 0 errors.
Failing this step because the tests timed out. Thread dumps were taken and archived here: https://github.com/apache/kafka/actions/runs/10850348751/artifacts/1930070177
Error: Process completed with exit code 1.
```

The thread dump archive is also in the job summary https://github.com/apache/kafka/actions/runs/10850348751?pr=17181#summary-30111636589

--- a/.github/scripts/junit.py
+++ b/.github/scripts/junit.py
@@ -263,8 +263,8 @@ if __name__ == "__main__":
         logger.debug(f"Gradle command timed out. These are partial results!")
         logger.debug(summary)
         if thread_dump_url:
-            print("\nThe JUnit tests were cancelled due to a timeout. Thread dumps were generated before the job was cancelled. "
-                  "Download [thread dumps]({thread_dump_url}).\n")
+            print(f"\nThe JUnit tests were cancelled due to a timeout. Thread dumps were generated before the job was cancelled. "
+                  f"Download [thread dumps]({thread_dump_url}).\n")
             logger.debug(f"Failing this step because the tests timed out. Thread dumps were taken and archived here: {thread_dump_url}")
         else:
             logger.debug(f"Failing this step because the tests timed out. Thread dumps were not archived, check logs in JUnit step.")

